examples/stable_diffusion_v2/T2I-Adapter.md

```diff
@@ -194,9 +250,40 @@ python examples/stable_diffusion_v2/adapter_image2image.py \
 --ddim
 ```
 
+</details>
+
 ## Training
 
-Coming soon.
+### Data preparation
+
+Conditional images must be in RGB format. Therefore, some datasets may require preprocessing before training.
+
+#### Segmentation (COCO-Stuff)
+
+Segmentation masks are usually grayscale images with values corresponding to class labels. However, to train
+T2I-Adapters, masks must be converted to RGB images. To do so, first download grayscale masks
+([stuffthingmaps_trainval2017.zip](https://github.com/nightrome/cocostuff#downloads)) and unpack them. Then execute the
+following command to convert mask to RGB images:
+
+```shell
+python examples/stable_diffusion_v2/tools/data_utils/t2i_adapter/cocostuff_colorize_mask.py PATH_TO_GRAY_MASKS_DIR PATH_TO_OUTPUT_DIR
+```
+
+Annotation labels for COCO-Stuff can be found in
+[annotations_trainval2017.zip/annotations/](https://github.com/nightrome/cocostuff#downloads).
+
+### Segmentation
+
+After the data preparation is completed, the following command can be used to train T2I-Adapter:
+
+```shell
+mpirun --allow-run-as-root -n 4 python examples/stable_diffusion_v2/train_t2i_adapter.py \
+--config examples/stable_diffusion_v2/configs/t2i-adapter/v2.1_train.yaml \
+--train.dataset.init_args.image_dir PATH_TO_IMAGES_DIR \
+--train.dataset.init_args.masks_path PATH_TO_RGB_MASKS_DIR \
+--train.dataset.init_args.label_path PATH_TO_LABELS \
+--train.output_dir: PATH_TO_OUTPUT_DIR
+```
```

Comment:
Please add Evaluation section here to show the generation samples and metrics  for the training adapter.

Reply:
Done